app/src/main/java/com/coveros/coverosmobileapp/blogpost/CommentFormActivity.java

```diff
@@ -65,30 +78,111 @@ public void onClick(View v) {
         return emptyFields;
     }
 
-    private AlertDialog createEmptyFieldAlertDialog(List<String> emptyFields) {
-        emptyFieldAlertDialog = new AlertDialog.Builder(CommentFormActivity.this).create();
-        emptyFieldAlertDialog.setTitle(R.string.empty_field_alert_dialog_title);
-        String emptyFieldsString;
-        if (emptyFields.size() == 1) {
-            emptyFieldsString = emptyFields.get(0);
-        } else if (emptyFields.size() == 2) {
-            emptyFieldsString = emptyFields.get(0) + " and " + emptyFields.get(1);
-        } else {
-            emptyFieldsString = emptyFields.get(0) + ", " + emptyFields.get(1) + ", and " + emptyFields.get(2);
+    class SendButtonOnClickListener implements View.OnClickListener {
+        @Override
+        public void onClick(View v) {
+            author = ((EditText) findViewById(R.id.enter_name)).getText().toString();
+            email = ((EditText) findViewById(R.id.enter_email)).getText().toString();
+            message = ((EditText) findViewById(R.id.enter_message)).getText().toString();
+            List<String> emptyFields = checkFieldIsEmpty(author, email, message);
+
+            final Pattern emailPattern = Pattern.compile("(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|\"(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21\\x23-\\x5b\\x5d-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])*\")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21-\\x5a\\x53-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])+)\\])");
+            Matcher emailMatcher = emailPattern.matcher(email);
+            boolean isValidEmail = emailMatcher.matches();
+
+            JsonObject body = createCommentRequestBody(postId, author, email, message);
+
+            commentRequest = new RestRequest(COMMENT_URL, null, body, new Response.Listener() {
+                @Override
+                public void onResponse(Object response) {
+                    createSuccessDialog(CommentFormActivity.this);
+                    successDialog.show();
+                }
+            }, new Response.ErrorListener() {
+                @Override
+                public void onErrorResponse(VolleyError error) {
+                    createErrorDialog(CommentFormActivity.this);
+                    errorDialog.show();
+                }
+            });
+
+            boolean hasNoEmptyFields = emptyFields.isEmpty();
+            if (hasNoEmptyFields && isValidEmail) {
+                RequestQueue requestQueue = Volley.newRequestQueue(CommentFormActivity.this);
+                requestQueue.add(commentRequest);
+            } else if (!isFinishing()) {
+                if (!hasNoEmptyFields) {
```

Comment:
Can you refactor this to avoid the double negation? ("does not have no empty fields" is  more difficult to understand than "has empty fields".)